### PR TITLE
Add support for reading fans on Aquaero

### PR DESCRIPTION
**Pull request by leoratte**
Adds support for reading fan speed, voltage, power, and current.
I added a new struct `fan_structure_offsets` because the values for each fan for Aquaero are in a different order than the other devices.

```
aquaero-hid-3-9
Adapter: HID adapter
Fan 1 voltage:  11.86 V  
Fan 2 voltage:  11.95 V  
Fan 3 voltage:  12.00 V  
Fan 4 voltage:  12.03 V  
Fan 1 speed:   1368 RPM
Fan 2 speed:      0 RPM
Fan 3 speed:   1457 RPM
Fan 4 speed:      0 RPM
Sensor 1:       +27.1°C  
Sensor 2:           N/A  
Sensor 3:       +27.5°C  
Sensor 4:           N/A  
Sensor 5:           N/A  
Sensor 6:       +26.8°C  
Sensor 7:           N/A  
Sensor 8:       +27.0°C  
Fan 1 power:     1.37 W  
Fan 2 power:     0.00 W  
Fan 3 power:     1.39 W  
Fan 4 power:     0.00 W  
Fan 1 current: 116.00 mA 
Fan 2 current:   0.00 A  
Fan 3 current: 116.00 mA 
Fan 4 current:   0.00 A  
```

### CODE REVIEW

File: aquacomputer_d5next.c [hunk 1]
```diff
@@ -211,6 +218,29 @@ static const char *const label_fan_flow_speed_quadro[] = {
...
+
+/* Fan structure offsets for Aquaero */
+static struct fan_structure_offsets fan_structure_aquaero = {
+	.voltage = AQUAERO_FAN_VOLTAGE_OFFSET,
+	.ampere = AQUAERO_FAN_CURRENT_OFFSET,
+	.power = AQUAERO_FAN_POWER_OFFSET,
+	.speed = AQUAERO_FAN_SPEED_OFFSET
+};
+
+/* Fan structure offsets for D5 Next, Octo and Quadro */
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:236]: Not only those, but all others except the Aquaero. I'd reword this to convey that Aquaero is the exception.

File: aquacomputer_d5next.c [hunk 2]
```diff
@@ -211,6 +218,29 @@ static const char *const label_fan_flow_speed_quadro[] = {
 	"Flow speed [l/h]"
 };
 
+struct fan_structure_offsets {
+	u8 voltage;
+	u8 ampere;
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:223]: Call this `current` to stay consistent with hwmon.

File: aquacomputer_d5next.c [hunk 3]
```diff
@@ -211,6 +218,29 @@ static const char *const label_fan_flow_speed_quadro[] = {
 	"Flow speed [l/h]"
 };
 
+struct fan_structure_offsets {
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:221]: `aqc_fan_structure_offsets`. It must be clear from which driver the struct is coming from.

File: aquacomputer_d5next.c [hunk 4]
```diff
@@ -211,6 +218,29 @@ static const char *const label_fan_flow_speed_quadro[] = {
...
+	u8 voltage;
+	u8 ampere;
+	u8 power;
+	u8 speed;
+};
+
+/* Fan structure offsets for Aquaero */
+static struct fan_structure_offsets fan_structure_aquaero = {
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:229]: I'd call this `aqc_aquaero_fan_structure`.

File: aquacomputer_d5next.c [hunk 5]
```diff
@@ -211,6 +218,29 @@ static const char *const label_fan_flow_speed_quadro[] = {
...
+	.voltage = AQUAERO_FAN_VOLTAGE_OFFSET,
+	.ampere = AQUAERO_FAN_CURRENT_OFFSET,
+	.power = AQUAERO_FAN_POWER_OFFSET,
+	.speed = AQUAERO_FAN_SPEED_OFFSET
+};
+
+/* Fan structure offsets for D5 Next, Octo and Quadro */
+static struct fan_structure_offsets fan_structure_normal = {
```
Comment (aleksamagicka) [inline aquacomputer_d5next.c:237]: And here `aqc_general_fan_structure` would convey the meaning better.

**Review by aleksamagicka — CHANGES_REQUESTED** [review]
Comments are mostly naming related, tested and works on Aquaero 6. Will you add a README for re-docs for Aquaero 5 as you've done with the other devices?

File: aquacomputer_d5next.c (hunk 2 above)
Comment (leoratte) [reply aquacomputer_d5next.c:223]: I could not use current as it is already in use by:
./arch/x86/include/asm/current.h:18:17: Fehler: Feld »get_current« als Funktion deklariert
   18 | #define current get_current()

Comment (aleksamagicka) [reply aquacomputer_d5next.c:223]: Ah I see, just tried it and you're right. `ampere` it is then.

**Comment by aleksamagicka** [comment]
`curr` is even better, thanks!
